src/components/clusters/wizards/osd/Networking/ClusterProxy.test.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Formik } from 'formik';
+
+import {
+  HTTP_PROXY_PLACEHOLDER,
+  HTTPS_PROXY_PLACEHOLDER,
+} from '~/components/clusters/common/networkingConstants';
+import { FieldId } from '~/components/clusters/wizards/osd/constants';
+import { render, screen } from '~/testUtils';
+
+import { ClusterProxy } from './ClusterProxy';
+
+jest.mock('@patternfly/react-core', () => ({
+  ...jest.requireActual('@patternfly/react-core'),
+  useWizardContext: jest.fn(() => ({
+    goToStepByName: jest.fn(),
+    activeStep: { name: 'mockStepId' },
+  })),
+}));
+
+const buildTestComponent = (formValues = {}) => (
+  <Formik
+    initialValues={{
+      [FieldId.HttpProxyUrl]: '',
+      [FieldId.HttpsProxyUrl]: '',
+      [FieldId.NoProxyDomains]: '',
+      [FieldId.AdditionalTrustBundle]: '',
+      ...formValues,
+    }}
+    onSubmit={jest.fn()}
+  >
+    <ClusterProxy />
+  </Formik>
+);
+
+describe('<ClusterProxy />', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('https URL validation fails when entering http URL', async () => {
+    const { user } = render(buildTestComponent());
+
+    const textBox = screen.getByPlaceholderText(HTTPS_PROXY_PLACEHOLDER);
+    await user.type(textBox, 'http://example.com');
+    await user.click(screen.getByPlaceholderText(HTTP_PROXY_PLACEHOLDER));
+
+    expect(
+      screen.getByText(/The URL should include the scheme prefix \(https:\/\/\)/i),
```

Comment:
you can avoid all the headache-inducing backslack escapes, if you replace the regex with a string and state `exact: false` in the options.

(`exact` controls the matcher [precision][1], and toggles case-sensitive and sub-string matching)



[1]: https://testing-library.com/docs/queries/about#precision